src/lib/registration/ui/addon_selection_base_dialog.rb

```diff
@@ -211,7 +249,8 @@ def handle_addon_selection(id)
         return unless addon
 
         show_addon_details(addon)
-        if Yast::UI.QueryWidget(Id(addon_widget_id(addon)), :Value)
+        new_state = ! addon.selected?
+        if new_state
```

Comment:
why not simple `addon.selected? ? addon.unselected : addon.selected` or adding it to addon class with method like toggle_selected?